server/service.go

```diff
@@ -164,32 +237,10 @@ func (m *BoostService) handleRegisterValidator(w http.ResponseWriter, req *http.
 		}
 	}
 
-	numSuccessRequestsToRelay := 0
-	var mu sync.Mutex
-
-	// Call the relays
-	var wg sync.WaitGroup
-	for _, relay := range m.relays {
-		wg.Add(1)
-		go func(relayAddr string) {
-			defer wg.Done()
-			url := relayAddr + pathRegisterValidator
-			log := log.WithField("url", url)
-
-			err := SendHTTPRequest(context.Background(), m.httpClient, http.MethodPost, url, payload, nil)
-			if err != nil {
-				log.WithError(err).Warn("error in registerValidator to relay")
-				return
-			}
-
-			mu.Lock()
-			defer mu.Unlock()
-			numSuccessRequestsToRelay++
-		}(relay.Address)
-	}
-
-	// Wait for all requests to complete...
-	wg.Wait()
+	// Send the payload to the goroutine responsible for handling the resend at interval
+	m.newRegistrationsRequests <- payload
+	// Block until we get the number of successful requests back from this goroutine
+	numSuccessRequestsToRelay := <-m.numSuccessRequestsToRelay
```

Comment:
Once context (timeout) handling is implemented this will become a race, the result channel (or callback) should be specific to the request.
I think pushing a pair of (payload, result chan) would be a clean way to solve it that will also work nicely with context timeouts.